source/base/Allocator.ooc

```diff
@@ -0,0 +1,25 @@
+/* This file is part of magic-sdk, an sdk for the open source programming language magic.
+ *
+ * Copyright (C) 2016 magic-lang
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+import Owner
+
+AbstractAllocator: abstract class {
```

Comment:
Another way is to simply have a concrete `Allocator` class where you can supply allocate/deallocate closures. That way you don't have to inherit the class for every single allocator, only if you need to add more behaviour. Also, then we can add static default allocators in this class.

```ooc
Allocator: class 
malloc := static This new( func { malloc ... }, func { free... })
```